Rubberduck.Refactorings/Common/ModuleBodyElementDeclarationExtensions.cs

```diff
@@ -0,0 +1,176 @@
+﻿using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Rubberduck.Refactorings.Common
+{
+    public static class ModuleBodyElementDeclarationExtensions
+    {
+        /// <summary>
+        /// Returns ModuleBodyElementDeclaration signature with an ImprovedArgument list
+        /// 1. Explicitly declares Property Let\Set value parameter as ByVal
+        /// 2. Ensures UserDefined Type parameters are declared either explicitly or implicitly as ByRef
+        /// </summary>
+        /// <param name="declaration"></param>
+        /// <returns></returns>
+        public static string FullMemberSignature(this ModuleBodyElementDeclaration declaration,
```

Comment:
:+1: for not extending `Declaration` (and thus not hiding the coupling) - note, there's something funny with extension methods that extend a specific concrete implementation: normally you'd write extension methods against an interface (and treat all implementations equally).
I guess what I'm trying to say is, there's probably a better design, but I can't think of one at the moment, and this works, so... moving on =)